pkg/api/types.go

```diff
@@ -1664,6 +1667,104 @@ type SecretList struct {
+	// and volumes
+	SELinuxOptions
+}
+
+// SELinuxOptions are the labels to be applied to the container
+type SELinuxOptions struct {
+	// User --security-opt="label:user:USER"
+	User string
+
+	// Role --security-opt="label:role:ROLE"
+	Role string
+
+	// Type --security-opt="label:type:TYPE"
+	Type string
+
+	// Level --security-opt="label:level:LEVEL"
+	Level string
+
+	// Disabled --security-opt="label:disable"
+	Disabled bool
+}
+
+// SecurityConstraints provides the constraints that at security context provider will
+// ensure that applied SecurityContext requests follow.  When a setting is provided in the SecurityConstraints
+// that conflicts with an actual request it will be implementation specific whether that request is
+// ignored and the container is still run (without the requested constraint) or if the container will be failed
+type SecurityConstraints struct {
+	// EnforcementPolicy will drive behavior for how the constraints are enforce
+	EnforcementPolicy SecurityConstraintPolicy
+
+	// AllowPrivileged indicates whether this context allows privileged mode containers
+	AllowPrivileged bool
+
+	// SELinux provides the security constraint options for selinux
+	SELinux SELinuxSecurityConstraints
+
+	// AllowCapabilities dictates if a container can request to add or drop capabilites
+	AllowCapabilities bool
+
+	// Capabilities represents, if AllowCapabilities is true, the caps that requests
+	// are allowed to add or drop.
+	Capabilities Capabilities
+
+	// DefaultSecurityContext is applied to any container that does not have a security context set.  It must
+	// also conform to the constraints defined in SecurityConstraints object
+	DefaultSecurityContext SecurityContext
+}
+
+// SELinuxSecurityConstraints defines what is allowed in SecurityContext requests with regards to SELinux label options
+// that are currently supported by docker
+type SELinuxSecurityConstraints struct {
+	// AllowUserLabel --security-opt="label:user:USER"
+	AllowUserLabel  bool
+
+	// AllowRoleLabel --security-opt="label:role:ROLE"
+	AllowRoleLabel bool
+
+	// AllowTypeLabel --security-opt="label:type:TYPE"
+	AllowTypeLabel  bool
+
+	// AllowLevelLabel --security-opt="label:level:LEVEL"
+	AllowLevelLabel bool
+
+	// AllowDisable --security-opt="label:disable"
+	AllowDisable    bool
+}
+
+// SecurityConstraintPolicy dictates how the security context provider should behave with regards to contexts that
+// do not meet the requirements of the policy.
+type SecurityConstraintPolicy string
+
+const (
+	// SecurityConstraintPolicyDisabled means that any containers that do not meet policy constraints are still
+	// allowed to run
+	SecurityConstraintPolicyDisabled = "Disabled"
+
+	// SecurityConstraintPolicyPermissive means that any containers that do not meet policy constraints will be
+	// modified to meet the constraints (settings will be changed) before they are run
+	SecurityConstraintPolicyPermissive = "Permissive"
```

Comment:
At this point the definition of permissive/restrict in this code is: check against defined policy, manipulate to fit within constraints.  So, no, there is no explicit permissive although we could make disabled work like SELinux permissive (check, warn, still run).

To me the question is still: is there a valid use case for the restrict option or is it just going to cause issues down the road?  I think it is likely that if a container is requesting escalated privileges then it needs them to run correctly.